src/memdb/memdb.ts

```diff
@@ -7,20 +11,88 @@ import {
   NewClientOptions,
 } from '../plugin/plugin.js';
 import { sync } from '../scheduler/scheduler.js';
-import { Table, createTable, filterTables } from '../schema/table.js';
+import { Table, createTable, filterTables, decodeTable, decodeRecord, getPrimaryKeys } from '../schema/table.js';
 
 export const createMemDBClient = () => {
-  return { id: () => 'memdb' };
+  //eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const memoryDB: Record<string, any[]> = {};
+  const tables: Record<string, Table> = {};
+  return {
+    id: () => 'memdb',
+    memoryDB,
+    tables,
+  };
 };
 
 export const newMemDBPlugin = (): Plugin => {
   const memdbClient = createMemDBClient();
+  const memoryDB = memdbClient.memoryDB;
+  const tables = memdbClient.tables;
 
   const allTables: Table[] = [
     createTable({ name: 'table1', title: 'Table 1', description: 'Table 1 description' }),
     createTable({ name: 'table2', title: 'Table 2', description: 'Table 2 description' }),
   ];
 
+  //eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const memdb: { inserts: any[]; [key: string]: any } = {
```

Comment:
```suggestion
  const memdb: { inserts: unknown[]; [key: string]: unknown } = {
```

This could work I think